### PR TITLE
Refactor make cogs

### PR DESCRIPTION
**Pull request by jhkennedy**
* Remove overviews
* `make_cog`'s API has change, but it's only used internally to `make_cogs.py`: https://github.com/search?q=org%3Aasfadmin+make_cog&type=Code
* `cogify_dir`'s API has also change, but it's only used by `hyp3-rtc-gamma` and `hyp3-rtc-snap`: https://github.com/search?q=org%3Aasfadmin+cogify_dir&type=Code
  * Opened asfadmin/hyp3-rtc-snap#75 and asfadmin/hyp3-rtc-gamma#113
* Also drop `repackage_into_cogs.sh` as it's not used anywhere: https://github.com/search?q=org%3Aasfadmin+repackage_into_cogs&type=Code

### CODE REVIEW

**Comment by jhkennedy** [comment]
We could probably also test this with the GeoTIFF fixture we added...

**Comment by asjohnston-asf** [comment]
Code looks fine.  I agree testing is do-able.  test_geotiff.tif is *already* a COG.  We could change it to be un-tiled and un-compressed, that wouldn't impact test_make_browse, but would let us verify make_cogs.  https://github.com/rouault/cog_validator is an option for doing the validation, but that might be a little heavyweight for our case.

File: hyp3lib/make_cogs.py [hunk 1]
```diff
@@ -1,45 +1,41 @@
...
-def cogify_dir(dir="PRODUCT",debug=False,res=30):
-    back = os.getcwd()
-    os.chdir(dir)
-    tmpfile = "tmp_cog_{}.tif".format(os.getpid())
-    for myfile in glob.glob("*.tif"):
-        logging.info("Converting file {} into COG".format(myfile))
-        make_cog(myfile,tmpfile,res=res)
-        shutil.move(tmpfile,myfile)
-    os.chdir(back)
+def cogify_dir(directory: str = "PRODUCT", file_pattern: str = '*.tif'):
```
Comment (asjohnston-asf) [inline hyp3lib/make_cogs.py:14]: now that I think about it, `directory` really shouldn't have a default value.  I'm thinking of the poor developer encountering a line of code in the wild that's just `cogify_dir()`; no reason to expect that's cogifying the `PRODUCT` directory

File: hyp3lib/make_cogs.py [hunk 2]
```diff
@@ -1,45 +1,41 @@
...
+    logging.info(f'Converting files to COGs for {path_expression}')
+    for filename in glob(path_expression):
+        make_cog(filename)
 
 
-def make_cog(inFile,outFile,debug=False,res=30):
-    print("Creating COG file {} from input file {}".format(outFile, inFile))
-    tmpFile = 'cog_{}.tif'.format(os.getpid())
-    shutil.copy(inFile,tmpFile)
+def make_cog(filename: str):
```
Comment (asjohnston-asf) [inline hyp3lib/make_cogs.py:27]: and while we're making breaking changes this might as well be `cogify_file`

File: hyp3lib/make_cogs.py (hunk 1 above)
Comment (asjohnston-asf) [reply hyp3lib/make_cogs.py:14]: or maybe `directory='.'` or `directory=''` would be better, that would at least cog-ify the working directory

Comment (asjohnston-asf) [reply hyp3lib/make_cogs.py:14]: my approval still stands, though, feel like we have have revised this PR to death already

File: hyp3lib/make_cogs.py (hunk 2 above)
Comment (jhkennedy) [reply hyp3lib/make_cogs.py]: I agree; done

File: hyp3lib/make_cogs.py (hunk 1 above)
Comment (jhkennedy) [reply hyp3lib/make_cogs.py]: I agree with no default actually. I think as a library function you should have to be explicit

**Pull request merged**